### ISSUE TITLE
pull-kubernetes-e2e-kops-aws is flaky

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#46370]
The pull-kubernetes-e2e-kops-aws suite has been very flaky recently. Sometimes nodes don't come up during cluster setup, sometimes tests failed even though the change was trivial. It seems to be much more unstable than other suites.

Here are recent failures:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/46309/pull-kubernetes-e2e-kops-aws/26529/
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/46363/pull-kubernetes-e2e-kops-aws/26733/
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/46242/pull-kubernetes-e2e-kops-aws/26731/
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/46132/pull-kubernetes-e2e-kops-aws/26703/
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/45918/pull-kubernetes-e2e-kops-aws/26688/

**Comment by dchen1107** [comment]
pull-kubernetes-e2e-kops-aws is one of builds blocking the submit-queue mostly. I raised the priority for this issue; otherwise, we should remove this for pre-submit. 

cc/ @kubernetes/kubernetes-release-managers @kubernetes-sig-aws @justinsb

**Comment by justinsb** [comment]
Is there a source for numbers?

I'm seeing a lot of failing GCE tests, but I _think_ we should keep GCE in the test suite ;-)

**Comment by justinsb** [comment]
More seriously... the first few seem to be the DisruptionController flakes coming back.

The ones where kops times out, I haven't tracked down yet.

We probably should start by changing:

`W0524 08:38:49.867] 2017/05/24 08:38:49 e2e.go:417: kubectl get nodes failed: "exit status 1": ""` so it doesn't swallow the actual error message..

**Comment by justinsb** [comment]
Aha ... found the numbers here: https://storage.googleapis.com/k8s-metrics/flakes-latest.json

So pull-kubernetes-unit is actually the worst.  pull-kubernetes-e2e-gce-etcd3 is the worst of the main tests, though pull-kubernetes-e2e-kops-aws is next up.

@dchen1107 can you assign some google resources to assist in our figuring out what is wrong here?  With the current test setup, it is hard to do so outside of google.

**Comment by dchen1107** [comment]
pull-kubernetes-e2e-gce-etcd3 was completely broken due to the misconfigured test infrastructure #46713 which should be fixed by kubernetes/test-infra#2902.

**Comment by justinsb** [comment]
Started submitting the changes needed for diagnostics https://github.com/kubernetes/test-infra/pull/2914

**Comment by yujuhong** [comment]
> Aha ... found the numbers here: https://storage.googleapis.com/k8s-metrics/flakes-latest.json
> 
> So pull-kubernetes-unit is actually the worst. pull-kubernetes-e2e-gce-etcd3 is the worst of the main tests, though pull-kubernetes-e2e-kops-aws is next up.

I didn't find any existing issue for the failures I saw, so I filed this broad issue as a catch all. In no way I was trying to indicate this was the only flaky suite :)

FYI, the bot created #46740 for the unit test failures, but most if not all the failures listed there were already addressed in separate issues.

**Comment by justinsb** [comment]
Cool, yeah sorry for over-reacting!  I was helping with other tests that the test-infra had identified as higher priority (e.g. GCE multizone), so it came as a bit of a surprise that "my" house was not in order!

I am working to track down the issues where kops fails to come up in the timeout.  Finding & fixing lots of stuff, but no smoking gun yet!

**Comment by falenn** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  curtis.l.bates@gmail.com
    Domain imwiz.com has exceeded the max emails per hour (155/150 (103%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o6.sgmail.github.com ([192.254.113.101]:32530)
	by box969.bluehost.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.87)
	(envelope-from <bounces+848413-5c7e-dev=imwiz.com@sgmail.github.com>)
	id 1dGTKt-000TlQ-0L
	for dev@imwiz.com; Thu, 01 Jun 2017 10:52:36 -0600
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=HvSvoiM2MV0adEj32BudXRzdR5M=; b=WzLxqU+zYiYz9nZR
	CO98zGZ3v0Y40+2yTZ8lFlJNOn5Z4EoJ2d2miAjdCLGD3PTQjWFOC7KMx0dYtCi2
	Wyt8Usv4RtQuOJkHWdFGWETKgRsdGSBP2StP7z3yUVFimduLTO9Io8xJQAEoimgY
	CwH8WFv/dkXUU+2PpHHzEHqxhXQ=
Received: by filter1174p1mdw1.sendgrid.net with SMTP id filter1174p1mdw1-20688-59304648-8
        2017-06-01 16:52:24.126199843 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0003p1iad1.sendgrid.net (SG) with ESMTP id 3VYQGCgyR3WAVtERFVzuNA
	for <dev@imwiz.com>; Thu, 01 Jun 2017 16:52:24.056 +0000 (UTC)
Date: Thu, 01 Jun 2017 09:52:23 -0700

**Comment by dchen1107** [comment]
@justinsb @krzyzacy is going to help you looking into test-infra related issues. Thanks!

**Comment by justinsb** [comment]
Dug into two tests, one successful, one failing, and compared them.  Opened #46871 with what I have found.  kube-scheduler stops logging.

Going to see if I can reproduce locally testing 1.7 / 1.8.  I haven't seen this with 1.6, so it might be something in 1.7, but then I don't understand why it wouldn't affect all tests.

**Comment by mbohlool** [comment]
this is failing for #46734 not sure if it is the same issue.

**Comment by justinsb** [comment]
I am treating this as blocked on #46871 ; it appears that the majority of the flakes are a real issue with a k-c-m / resource race, which kops is more likely to encounter than kube-up for some reason.  (I suspect the reason is that kops bringup is more parallel)

**Comment by MikeSpreitzer** [comment]
Why is this issue not labelled `kind/flake` ?

**Comment by justinsb** [comment]
@MikeSpreitzer I think because it was not opened by the test infra.

Propose closing; the root cause has been fixed, and we should let test-infra open if needed.
